### PR TITLE
Initial pass on Python API

### PR DESCRIPTION
**Pull request by JeanArhancet**
This pull request introduces the initial setup for the Python bindings (#248).

- Setup Configuration: Added the Python binding stack, including the `pyo3 `crates, `pyproject.toml`, `build.rs`, and other necessary files.
- Database Class: Implemented the Database class with a constructor to establish a connection and a query function to execute SQL queries.
- Testing: Created `database.db` with a sample users table and two entries, as outlined in README.md, and added three pytest functions to validate the Python output.

### CODE REVIEW

**Comment by penberg** [comment]
Let's make sure the API follows PEP 249 (https://peps.python.org/pep-0249/), preferably so that it's a drop-in replacement for the sqlite3 API in Python: https://docs.python.org/3/library/sqlite3.html

**Comment by JeanArhancet** [comment]
> Let's make sure the API follows PEP 249 (https://peps.python.org/pep-0249/), preferably so that it's a drop-in replacement for the sqlite3 API in Python: https://docs.python.org/3/library/sqlite3.html

@penberg I've refactored the code to comply with PEP 249. What do you think?

File: bindings/python/tests/test_database.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    cursor = conn.cursor()
+    cursor.execute("SELECT * FROM users")
+
+    users = cursor.fetchall()
+    assert users
+    assert users == [(1, "alice"), (2, "bob")]
+
+
+def test_sqlite3_fetchall_select_all_users():
+    conn = sqlite3.connect("tests/database.db")
```
Comment (penberg) [inline bindings/python/tests/test_database.py:16]: No need to change this now, but you can use @pytest annotation to parametrize a _same_ test case to run with both libraries. See, for example, https://github.com/tursodatabase/libsql-experimental-python/blob/main/tests/test_suite.py

**Comment by penberg** [comment]
Looks great @JeanArhancet! All we need to figure out is how to get this to build on Windows... 🙈

File: bindings/python/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+        .into()
+    }
+}
+
+#[pyclass]
+pub struct Cursor {
+    /**
+     * This read/write attribute specifies the number of rows to fetch at a time with .fetchmany().
+     * It defaults to 1 meaning to fetch a single row at a time.
+     */
```
Comment (penberg) [inline bindings/python/src/lib.rs:53]: Btw, this is _not_ rustdoc style, see https://doc.rust-lang.org/rust-by-example/meta/doc.html for examples.

File: bindings/python/tests/test_database.py (hunk 1 above)
Comment (JeanArhancet) [reply bindings/python/tests/test_database.py]: Fixed

**Pull request merged**